transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -230,11 +232,16 @@ private void doReleaseChannel(Channel channel, Promise<Void> promise) {
                          promise);
         } else {
             try {
-                if (offerChannel(channel)) {
-                    handler.channelReleased(channel);
-                    promise.setSuccess(null);
+                final Future<Boolean> f = healthCheck.isHealthy(channel);
```

Comment:
As discussed in https://github.com/netty/netty/issues/4077#issuecomment-130461684. Perhaps we should make this check configurable (some boolean flag to indicate we should check health when putting back into pool).